src/compiler/types.ts

```diff
@@ -4966,6 +4966,14 @@ export interface EmitTransformers {
     declarationTransformers: readonly TransformerFactory<SourceFile | Bundle>[];
 }
 
+/** @internal */
+
+export interface DeclarationTransformer {

```

Comment:
I'm confused why we need this; is `Transformer<SourceFile | Bundle>` sufficient, like our other transforms?

Reply:
Someone who is more familiar with these transforms should weigh in here

It didn't seem to me that `Transformer<SourceFile | Bundle>` would be sufficient, because it seemed to be important here that we get the same and narrowest type out of the transform as the type put in. At least some of the other `Transformer<SourceFile | Bundle>` could/should be typed conditionally (if we supported it) so that the input is the same type as the output, and I don't think it's possible to currently improve on those function types without also adding overloads.

Reply:
I guess the existing function already has the overload; honestly I'd just move the type to the file where it's used. But, I changed the signature to:

```
function transformRoot(node: SourceFile | Bundle): SourceFile | Bundle {
```

And got no compiler errors, so I don't think it's actually critical.